chainforge/react-server/src/PromptNode.js

```diff
@@ -691,6 +716,7 @@ const PromptNode = ({ data, id, type: node_type }) => {
             } else {
                 rejected(json.error || 'Unknown error when querying LLM');
             }
+            QueryTracker.delete(id);
```

Comment:
Since there's a return on line 667, there's a case where this will never trigger. I also think that this error should be dealt with in `rejected` function --where you check the type of the error --instead of here. That is because, as you have coded it, you seem to raise an Error upon cancellation. That will in turn reject the Promise producted by `queryllm`, which should then bubble up to `rejected` here. 

At least, it should. Let me know if that's not the case.

Reply:
I didn't bubble it up all the way. Once the error bubbles to queryLLM, I returned undefined instead so that it will always hit the case if(!json)